### PR TITLE
Fix #25626: Favorite Pages Unify Relative Date Calculation To One Single Pipe

### LINKED ISSUE #25626: Favorite Pages: 'Last Edited' date displays incorrectly when editing or creating any file
### Parent Issue

[Pages]

### Problem Statement

When editing or creating any file, the last edit date does not display correctly.

### Steps to Reproduce

1. Go to https://localhost:8443/dotAdmin/#/pages
2. Create or edit a file
3. Observe that the displayed last edit date does not reflect the actual date/time of the edit

### Acceptance Criteria

When a file is edited or created, the last edit date should accurately reflect the date and time of the most recent modification.

### dotCMS Version

23.07

### Proposed Objective

User Experience

### Proposed Priority

Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

Screenshot 2023-07-26 at 9 17 27 AM

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by zJaaal**
### Note
Updated due to changes in the approach.

### Summary
🗑️🔄🕒

Refactored the `dotRelativeDate` pipe to handle UTC dates and added optional parameters for date formatting. Replaced the `dotTransformVersionLabel` pipe with the `dotRelativeDate` pipe and the `dm` pipe in the dot-content-compare component. Updated the templates that use the `dotRelativeDate` pipe to remove the unnecessary second argument. Deleted the unused `dotTransformVersionLabel` pipe and its spec file.

> _We've cleaned up the `dotRelativeDate` pipe, me hearties_
> _No more second argument to pass around_
> _It handles the date conversion on its own, me hearties_
> _And shows the proper format for every sound_

### Walkthrough
*  Updated the `DotRelativeDatePipe` to handle date conversion and formatting internally ([link](https://github.com/dotCMS/core/pull/25682/files?diff=unified&w=0#diff-276797d85550e814a2f1974b90e9f556c6b08c767d128bfc762945515e194fbdL12-R41))
*  Removed the second argument from the `dotRelativeDate` pipe in several templates, as it was no longer needed ([link](https://github.com/dotCMS/core/pull/25682/files?diff=unified&w=0#diff-04da8afcab5e4883bc097456e55afa6dcedc2bf85aa55facc134e3ccb4656968L81-R81), [link](https://github.com/dotCMS/core/pull/25682/files?diff=unified&w=0#diff-ce14bb21a1834f2bb62ae18a86c955b285337e3a34381ba1ce0f40439a35558fL66-R66), [link](https://github.com/dotCMS/core/pull/25682/files?diff=unified&w=0#diff-0b7825dfbd0afe950947a35b9a764ff4604ae63af445d41cf9d1b6e0df95b627L1-R1), [link](https://github.com/dotCMS/core/pull/25682/files?diff=unified&w=0#diff-ea2b8676ef31a6a408a47a34d997a8025aabd199db71851f86c3a9763f6946a3L118-R118), [link](https://github.com/dotCMS/core/pull/25682/files?diff=unified&w=0#diff-47ff8e7da16ff598a09d04b521a054c50ccbca38ae913fec184a9440695f8b3fL25-R28))
*  Replaced the `dotTransformVersionLabel` pipe with a combination of the `dotRelativeDate` pipe and the `dm` pipe in the `dot-content-compare-table.component.html` template, to show the version date and modifier user name ([link](https://github.com/dotCMS/core/pull/25682/files?diff=unified&w=0#diff-dc18b122d0634d32911152c8a5dbe044a51cc9c321d6ce663cea4ebbc2e5b774L19-R28))
*  Removed the `DotTransformVersionLabelPipe` from the `dot-content-compare.module.ts` file, as it was no longer used, and added the `DotRelativeDatePipe` to the imports and declarations ([link](https://github.com/dotCMS/core/pull/25682/files?diff=unified&w=0#diff-9303672d35455beda6f8f17aebeb3c1d73b6989d91c8220be5e136a783e9465aL21-R23), [link](https://github.com/dotCMS/core/pull/25682/files?diff=unified&w=0#diff-9303672d35455beda6f8f17aebeb3c1d73b6989d91c8220be5e136a783e9465aL31-R31), [link](https://github.com/dotCMS/core/pull/25682/files?diff=unified&w=0#diff-9303672d35455beda6f8f17aebeb3c1d73b6989d91c8220be5e136a783e9465aL46-R46))
*  Deleted the `dot-transform-version-label.pipe.spec.ts` and `dot-transform-version-label.pipe.ts` files, as they were no longer needed ([link](https://github.com/dotCMS/core/pull/25682/files?diff=unified&w=0#diff-854799795d55e665484fe844dc5989586db209c94b9e0654975781f22282bbe8), [link](https://github.com/dotCMS/core/pull/25682/files?diff=unified&w=0#diff-2bf9d7f92a700744016ed6469befb30df3acfc2449b2d905319a1a432b9bb21d))
*  Updated the comment for the `getRelative` method in the `dot-format-date-service.ts` file, to reflect that the dates must be in UTC format ([link](https://github.com/dotCMS/core/pull/25682/files?diff=unified&w=0#diff-b5e47471864037ea0768d31b120e8e5e4881949ecbb6d547f28e56c8bff213c3L108-R108))
*  Modified the `getRelative` method in the `dot-format-date-service.ts` file, to accept `Date` objects as parameters and use the `getUTC` method to convert them to UTC ([link](https://github.com/dotCMS/core/pull/25682/files?diff=unified&w=0#diff-b5e47471864037ea0768d31b120e8e5e4881949ecbb6d547f28e56c8bff213c3L115-R132))
*  Added the `getUTC` method to the `dot-format-date-service.ts` file, to handle the date conversion logic ([link](https://github.com/dotCMS/core/pull/25682/files?diff=unified&w=0#diff-b5e47471864037ea0768d31b120e8e5e4881949ecbb6d547f28e56c8bff213c3L115-R132))

### Screenshots
https://github.com/dotCMS/core/assets/63567962/5ab3ceba-ee45-4f8c-b510-bf35429416ba

### CODE REVIEW

**Review by oidacra — APPROVED** [review]

**Review by manuelrojas — APPROVED** [review]

**Review by hmoreras — APPROVED** [review]

**Comment by zJaaal** [comment]
Not sure about this PR will wait for @fmontes to be sure.

File: core-web/apps/dotcms-ui/src/app/view/pipes/dot-relative-date/dot-relative-date.pipe.spec.ts [hunk 1]
```diff
@@ -1,21 +1,160 @@
...
+
+        it('should call dotFormatService.getUTC 3 times when its not a timestamp', () => {
+            const date = new Date();
+
+            spyOn(formatDateService, 'getUTC').and.callThrough();
+
+            pipe.transform(date.getTime()); // This is milliseconds not a timestamp
+
+            expect(formatDateService.getUTC).toHaveBeenCalledTimes(3);
+        });
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/view/pipes/dot-relative-date/dot-relative-date.pipe.spec.ts:68]: I feel we don't need to tests this, just need to make sure that the transform is returning what we expect.

File: core-web/apps/dotcms-ui/src/app/view/pipes/dot-relative-date/dot-relative-date.pipe.spec.ts [hunk 2]
```diff
@@ -1,21 +1,160 @@
...
+            spyOn(formatDateService, 'getUTC').and.callThrough();
+
+            pipe.transform(date); // This is a hardcoded timestamp
+
+            expect(formatDateService.getUTC).toHaveBeenCalledTimes(2); // It gets called twice in this workflow
+        }));
+    });
+
+    describe('format date', () => {
+        it('should return formated date after 7 days', fakeAsync(() => {
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/view/pipes/dot-relative-date/dot-relative-date.pipe.spec.ts:92]: Why `fakeAsync`?

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/apps/dotcms-ui/src/app/view/pipes/dot-relative-date/dot-relative-date.pipe.spec.ts (hunk 2 above)
Comment (zJaaal) [reply core-web/apps/dotcms-ui/src/app/view/pipes/dot-relative-date/dot-relative-date.pipe.spec.ts:92]: `tick()` only works under `fakeAsync`

File: core-web/apps/dotcms-ui/src/app/view/pipes/dot-relative-date/dot-relative-date.pipe.spec.ts (hunk 1 above)
Comment (zJaaal) [reply core-web/apps/dotcms-ui/src/app/view/pipes/dot-relative-date/dot-relative-date.pipe.spec.ts:68]: Alright! I'll delete it

**Review by rjvelazco — APPROVED** [review]

**Review by fmontes — APPROVED** [review]

**Pull request merged**